app/models/proxy_exercise.rb

```diff
@@ -37,30 +37,14 @@ def get_matching_exercise(user)
           assigned_user_proxy_exercise.exercise
         else
           matching_exercise =
-              if (token.eql? "e85689d5")
-                Rails.logger.debug("Proxy exercise with token e85689d5, split user in groups..")
-                group = UserGroupSeparator.getGroupExerciseDescriptionTesting(user)
-                Rails.logger.debug("user assigned to group #{group}")
-                case group
-                  when :group_a
-                    exercises.where(id: 557).first
-                  when :group_b
-                    exercises.where(id: 558).first
-                  when :group_c
-                    exercises.where(id: 559).first
-                  when :group_d
-                    exercises.where(id: 560).first
-                end
-              else
-                Rails.logger.debug("find new matching exercise for user #{user.id}" )
-                begin
-                  find_matching_exercise(user)
-                rescue => e #fallback
-                  Rails.logger.error("finding matching exercise failed. Fall back to random exercise! Error: #{$!}" )
-                  @reason[:reason] = "fallback because of error"
-                  @reason[:error] = "#{$!}:\n\t#{e.backtrace.join("\n\t")}"
-                  exercises.where("expected_difficulty > 1").shuffle.first # difficulty should be > 1 to prevent dummy exercise from being chosen.
-                end
+              Rails.logger.debug("find new matching exercise for user #{user.id}" )
```

Comment:
Does this actually work? (assigning matching_exercise) -> I think the log statement should move into the begin block. I fear otherwise you might be assigning whatever is returned by the logger to matching_exercise...